Firestore/core/test/firebase/firestore/util/status_test.cc

```diff
@@ -113,6 +113,43 @@ TEST(Status, FromErrno) {
           "Already exists: Cannot write file \\(errno .*: File exists\\)"));
 }
 
+TEST(Status, CausedBy_OK) {
+  Status result = Status::OK();
+  result.CausedBy(Status::OK());
+  EXPECT_EQ(Status::OK(), result);
+}
+
+TEST(Status, CausedBy_CauseOK) {
+  Status not_found(FirestoreErrorCode::NotFound, "file not found");
+
+  Status result = not_found;
+  result.CausedBy(Status::OK());
+  EXPECT_EQ(not_found, result);
+}
+
+TEST(Status, CausedBy_OuterOK) {
+  Status not_found(FirestoreErrorCode::NotFound, "file not found");
+
+  Status result = Status::OK();
+  result.CausedBy(not_found);
+  EXPECT_EQ(not_found, result);
+}
+
+TEST(Status, CausedBy_Chain) {
+  Status not_found(FirestoreErrorCode::NotFound, "file not found");
+  Status not_ready(FirestoreErrorCode::FailedPrecondition, "DB not ready");
+
+  Status result = not_ready;
+  result.CausedBy(not_found);
+  EXPECT_NE(not_found, result);
+  EXPECT_NE(not_ready, result);
+
+  // Outer should prevail
+  EXPECT_EQ(not_ready.code(), result.code());
+  EXPECT_EQ("Failed precondition: DB not ready: file not found",
```

Comment:
nit: this test verifies the exact formatting of CausedBy. A "better" test might be to verify all the individual components are present, and in the correct order.

No, I don't care either.  :)